src/containers/Chat/ChatMessages/ChatMessages.tsx

```diff
@@ -224,17 +225,19 @@ export const ChatMessages: React.SFC<ChatMessagesProps> = ({ contactId }) => {
       // allConversations.conversations.splice(0, 0, data.conversation);
       // allConversations.conversations.unshift(data.conversation);
     }
-  }
+  } else {
+    // Have contactId be first conversation contact id if there are conversations.
+    // If there are no conversations (new user), then return that there are "No conversations".
 
-  // By default, have conversationInfo be the first thing if there is no contactId.
-  // If there is no first conversation (new user), then return that there are "No conversations".
-  if (conversationInfo.length === 0) {
-    conversationIndex = 0;
-    // No conversations case
     if (allConversations.conversations.length === 0) {
       conversationInfo = null;
     } else {
-      conversationInfo = allConversations.conversations[conversationIndex];
+      // POTENTIAL FIX: If no contactId, redirect to URL with contactId.
+      // This could be on a higher component (App/Chat) instead of here. However, this is where we first know the contactId.
+      return <Redirect to={'/chat/'.concat(allConversations.conversations[0].contact.id)} />;
```

Comment:
This is currently my potential fix for this issue. After seeing that all these bugs are primarily focused on changing the chat URL from an empty contactId to an existing one, I thought about always redirecting to a conversation with a contactId instead. I also took a look at how other chat applications have done with related functionality (Messenger), and I realized that their URLs always redirect to the most recent conversation + contactId (similar to what I've implemented here).

I don't know if this is the best solution for this issue, but this + the additional fix on line 232 would ensure that there would never be a need for a user to enter the chat endpoint without a contactId and come across an error.

Reply:
Yes, that's fine. Let's do it.